### ISSUE TITLE
[BUG] wrong detection of mime parts within the body of a message/rfc822 if not base64 encoded 

### ISSUE THREAD

**Issue by agawish** [rspamd/rspamd#4456]
### Prerequisites

* [ X] Put an X between the brackets on this line if you have done all of the following:
    * Read about bug reporting in general: https://rspamd.com/doc/faq.html#how-to-report-bugs-found-in-rspamd
    * Enabled relevant debugging logs: https://rspamd.com/doc/faq.html#how-to-debug-some-module-in-rspamd 
    * Checked the FAQs about Core files in case of fatal crash: https://rspamd.com/doc/faq.html#how-to-figure-out-why-rspamd-process-crashed
    * Tried ASAN package and obtained the ASAN report (if possible): https://rspamd.com/doc/faq.html#asan-builds 
    * Checked that your issue isn't already filed: https://github.com/issues?utf8=%E2%9C%93&q=is%3Aissue+user%3Arspamd
    * Checked that there is not already an experimental package or master branch

**Describe the bug**

When RSPAMD parses an email with an `message/rfc822` part - that is not base64 - causes the parser to wrongly detect parts that are meant to be the body of `message/rfc822`.

### Steps to Reproduce
[o365-report.zip](https://github.com/rspamd/rspamd/files/11171023/o365-report.zip)

1. Download the zip file above
2. Run `rspamadm mime dump --json "$file"` pointing to the `eml` file
3. Validate the third part (below):
```
{
      "boundary": "_002_JMR04038c193f304aae8e5808db3071039ePhish638157037108144_",
      "headers": [
        {
          "decoded": "message/rfc822",
          "empty_separator": false,
          "name": "Content-Type",
          "order": 0,
          "raw": "Content-Type: message/rfc822\r\n",
          "separator": " ",
          "tab_separated": false,
          "value": "message/rfc822"
        },
        {
          "decoded": "attachment; creation-date=\"Wed, 01 Jan 1900 00:00:00 GMT\"; modification-date=\"Wed, 01 Jan 1900 00:00:00 GMT\"",
          "empty_separator": false,
          "name": "Content-Disposition",
          "order": 1,
          "raw": "Content-Disposition: attachment;\r\n\tcreation-date=\"Wed, 01 Jan 1900 00:00:00 GMT\";\r\n\tmodification-date=\"Wed, 01 Jan 1900 00:00:00 GMT\"\r\n",
          "separator": " ",
          "tab_separated": false,
          "value": "attachment; creation-date=\"Wed, 01 Jan 1900 00:00:00 GMT\"; modification-date=\"Wed, 01 Jan 1900 00:00:00 GMT\""
        }
      ],
      "size": 0,
      "type": "message/rfc822"
    },
```

**Expected behavior**
Expected the `message/rfc822` to include its content.

### Versions

Rspamd version 3.3
But I believe this would apply to all recent versions.

### Additional Information

This is a problem since it is a build in functionality within Microsoft Office 365 when they report a message as Phishing, so unfortunately it happens more times that it should have.

**Comment by vstakhov** [comment]
I cannot reproduce this issue. Here is my output of `mime dump`:

```
> rspamadm mime dump -j ~/Downloads/o365-report/o365-report.eml | jq '.parts[] | .type'

"multipart/mixed"
"text/plain"
"message/rfc822"
"multipart/mixed"
"text/plain"
"message/rfc822"
"multipart/alternative"
"text/plain"
"text/html"
```

So I can clearly see 2 `message/rfc822` parts there.

**Comment by agawish** [comment]
The content type detection is not wrong, but for example in the 3rd part, it should have put the full content of the the `message/rfc822` as a content rather than have 2 headers with `size:0`, Since this is an attachment, I should have seen as well `attachment: true`, so it looks like the parser detected a boundary and announced a new part even though from part 4 on ward it is just all part of the first `message/rfc822` and should have been just declared as a single part!

**Comment by vstakhov** [comment]
It seems I've lost the track. The structure of this message is quite hard to follow manually. I was under impression that you have provided the full output of `mime dump` whilst you haven't. I will investigate further.

**Comment by agawish** [comment]
The mime dump is in the zip file, I will try to make another easier example to follow. Sorry for the troubles.

**Comment by vstakhov** [comment]
Ok, it seems that message/rfc822 does not have it's content indeed. Is that a problem?

**Comment by vstakhov** [comment]
In fact, when I encode it's content with base64 I don't see any changes in the result.

**Comment by agawish** [comment]
You're correct, it is indeed treated the same, sorry for troubling, will close this ticket.

**Comment by agawish** [comment]
Not an issue, more of a user error!
